app/js/components/orderEntry/OrderEntryPage.jsx

```diff
@@ -1,17 +1,47 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
 import PatientDashboard from '../patientDashboard';
-
+import fetchPatientCareSetting from '../../actions/careSetting';
 import SearchAndAddOrder from './SearchAndAddOrder';
+import ActiveOrders from './ActiveOrders';
+
+export class OrderEntryPage extends React.Component {
+  componentDidMount() {
+    this.props.fetchPatientCareSetting();
+  }
 
-class OrderEntryPage extends React.Component {
   render() {
+    if (!(this.props.outpatientCareSetting || this.props.inpatientCareSetting)) {
+      return (
+        <div>
+          loading...
```

Comment:
@fuchodeveloper, I like the way you are using one endpoint for both careSettings, though the `<div>loading</div>` would look better visually say using a spinner than verbal.

Reply:
Will consult the team concerning the idea of using a reusable spinner in scenarios such as this and update this PR based on the team's response.

Reply:
Updated now to make use of the default OpenMRS spinner image based on the team's decision.